src/vs/workbench/parts/files/browser/views/explorerViewer.ts

```diff
@@ -524,8 +524,8 @@ export class FileController extends DefaultController {
 	}
 }
 
-// Explorer Sorter
-export class FileSorter implements ISorter {
+// Default Sorter
+export class DefaultSorter implements ISorter {
```

Comment:
There is lots of duplication across all sorters, specifically this code:

```typescript
if (statA instanceof NewStatPlaceholder) {
	return -1;
}

if (statB instanceof NewStatPlaceholder) {
	return 1;
}

// Do not sort roots
if (statA.isRoot) {
	return -1;
}

if (statB.isRoot) {
	return 1;
}
```

I suggest to reduce the number of filters to just one which listens to configuration changes and updates itself based on the config. That avoids the extra call to setSorter() in the tree and makes the sorter self contained based on the configuration.